sweep/README.md

```diff
@@ -0,0 +1,185 @@
+
+#### Sweeping Commit Outputs
+
+The only output we can spend is the `to_local` output. Because it can only be
+spent using our signature, there’s no time pressure here. By default, the
+sweeper will use a deadline of 1008 blocks as the confirmation target for
+non-time-sensitive outputs.  To overwrite the default, users can specify a
+value using the config `--sweeper.nodeadlineconftarget`.
+
+To specify the budget, users can use `--sweeper.budget.tolocal` to set the max
+allowed fees in sats, or use `--sweeper.budget.tolocalratio` to set a
+proportion of the `to_local` value to be used as the budget.
+
+#### Sweeping HTLC Outputs
+
+When facing a local force close transaction, HTLCs are spent in a two-stage
+setup - the first stage is to spend the outputs using pre-signed HTLC
+success/timeout transactions, the second stage is to spend the outputs from
+these success/timeout transactions. All these outputs are automatically handled
+by `lnd`, in specific,
+- For an incoming HTLC in stage one, the deadline is specified using its CLTV
+  from the timeout path. This output is time-sensitive.
+- For an outgoing HTLC in stage one, the deadline is derived from its
+  corresponding incoming HTLC’s CLTV. This output is time-sensitive.
+- For both incoming and outgoing HTLCs in stage two, because they can only be
+  spent by us, there is no time pressure to confirm them under a deadline.
+
+When facing a remote force close transaction, HTLCs can be directly spent from
+this transaction, and they are both time-sensitive outputs.
+
+By default, `lnd` will use 50% of the HTLC value as its budget. To customize
+it, users can specify `--sweeper.budget.deadlinehtlc` and
+`--sweeper.budget.deadlinehtlcratio` for time-sensitive HTLCs, and
+`--sweeper.budget.nodeadlinehtlc` and `--sweeper.budget.nodeadlinehtlcratio`
+for non-time-sensitive sweeps.
+
+#### Sweeping Anchor Outputs
+
+An anchor output is a special output that functions as “anchor” to speed up the
+unconfirmed force closing transaction via CPFP. Most of the time, anchor
+outputs won’t be swept due to it’s uneconomical. However, if the force close
+transaction contains time-sensitive outputs (HTLCs), and stays unconfirmed, the
+anchor output will be swept to accelerate the force close process.
+
+For CPFP-purpose anchor sweeping, its deadline is derived from the smallest
+deadline values of the HTLCs on the force close transaction. Its budget,
+however, is calculated differently - it cannot be a ratio of the anchor output
+because the value is too small (330 sats). Since its purpose is to accelerate
+the force close transaction so the time-sensitive outputs can be swept, the
+budget is in fact a ratio of what we call “value under protection”, which is
+the sum of all HTLC outputs minus sum of their budgets. By default, 50% of this
+value is used as the budget, to customize it, either use
+`--sweeper.budget.anchorcpfp` to specify sats, or use
+`--sweeper.budget.anchorcpfpratio` to specify a ratio.
```

Comment:
```suggestion
For CPFP-purpose anchor sweeping, the deadline is the closest
deadline value of all the HTLCs on the force close transaction. The budget,
however, cannot be a ratio of the anchor output
because the value is too small to contribute meaningful fees (330 sats). Since its purpose is to accelerate
the force close transaction so the time-sensitive outputs can be swept, the
budget is actually drawn from what we call “value under protection”, which is
the sum of all HTLC outputs minus the sum of their budgets. By default, 50% of this
value is used as the budget, to customize it, either use
`--sweeper.budget.anchorcpfp` to specify sats, or use
`--sweeper.budget.anchorcpfpratio` to specify a ratio.
```

Side note, I still think the behavior here is quite confusing.  Ideally the user just has to give one budget -- the deadline-aware budget, and commitment/HTLC fees would be automatically calculated.

Reply:
updated!

> the deadline-aware budget, and commitment/HTLC fees would be automatically calculated.

I like it - looks like I didn't get it before, we do need to figure how to allocate the budgets here tho, but I agree it makes `lnd` easier to use, probably should have an `advanced` config section.